### PR TITLE
treewide: Simplify negated uses of versionAtLeast, versionOlder

### PR DESCRIPTION
**Pull request by andersk**
###### Description of changes

Simplify `!lib.versionAtLeast v1 v2` to `lib.versionOlder v1 v2`, and `!lib.versionOlder v1 v2` to `lib.versionAtLeast v1 v2`.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [x] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/science/logic/coq/default.nix [hunk 1]
```diff
@@ -56,8 +56,8 @@ let
     args.version;
   version = fetched.version;
   coq-version = args.coq-version or (if version != "dev" then versions.majorMinor version else "dev");
-  versionAtLeast = v: (coq-version == "dev") || (lib.versionAtLeast coq-version v);
-  ideFlags = optionalString (buildIde && !versionAtLeast "8.10")
+  coqAtLeast = v: coq-version == "dev" || versionAtLeast coq-version v;
+  ideFlags = optionalString (buildIde && !coqAtLeast "8.10")
```
Comment (Mindavi) [inline pkgs/applications/science/logic/coq/default.nix:60]: For these I'm wondering why there's not both `coqAtLeast` and `coqOlder`, now there's still negated usage.

**Comment by Mindavi** [comment]
In general this seems like a good change, improving readability and setting a precedent for future changes.

File: pkgs/applications/science/logic/coq/default.nix (hunk 1 above)
Comment (andersk) [reply pkgs/applications/science/logic/coq/default.nix:60]: Adding `coqOlder` seems like a reasonable further cleanup. But there are other cleanups I’d want to do there too (like…what is up with these bizarre non-transitive [truncating version comparators](https://github.com/NixOS/nixpkgs/blob/87d34a6b8982e901b8e50096b8e79ebc0e66cda0/pkgs/build-support/coq/extra-lib.nix#L4-L68) that get [monkey-patched into `lib`](https://github.com/NixOS/nixpkgs/blob/87d34a6b8982e901b8e50096b8e79ebc0e66cda0/pkgs/applications/science/logic/coq/default.nix#L17)…this is what half-open ranges are for…?). So I’m inclined to defer that discussion to a future Coq-specific PR so as not to bother the other maintainers that got automatically pinged here.

**Review by SuperSandro2000 — APPROVED** [review]

File: pkgs/applications/science/logic/coq/default.nix (hunk 1 above)
Comment (Mindavi) [reply pkgs/applications/science/logic/coq/default.nix:60]: Makes sense.

**Review by Mindavi — APPROVED** [review]

**Pull request merged**